monoDriveIO/monodrive-client
# Opendrive automated vandv

## Overview

This automates many canary tests that help debug the simulator and client. The tests in the `diagnostics` directory can be run with ctest to quickly diagnose any errors or bottles necks in local and distributed systems.

Additionally, the response message from the sending server commands has been updated to include the entire `ApiMessage` data structure for a more clean means of parsing responses from the server.

## Changes 

* Implemented several server command/response tests to ensure connectivity with the simulator and supported functionality
* Implemented several sensor tests to validate sensor outputs and performance
* Modified `sendCommand` to respond with an `ApiMessage`
* Fixed some of the lane following examples to use new `sendCommand`

## Testing
* run all diagnostic test with `ctest --verbose` from your `./build` directory
* test that the client fails gracefully when the server is unlicensed
* general evaluate logging approach (open to other suggestions or patterns)
* make sure that examples fail loudly then the file path or working directory is incorrect
* lane_follower/fixed_step
* lane_follower/fixed_step_lidar
* lane_follower/real_time
* lane_follower/speed_curve
* multi_process_example/vehicle_controller.cpp
